mrp_warehouse_calendar/models/mrp_production.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2018-19 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    @api.depends("company_id", "date_planned_start", "is_planned", "product_id")
+    def _compute_date_planned_finished(self):
+        for production in self:
+            res = super(MrpProduction, self)._compute_date_planned_finished()
```

Comment:
you are creating a loop inside a loop here by calling super with all records inside the for.

you should call super first outside the loop, then iterate (ideally a filtered recordset) and update what's needed.